NEWS.md

```diff
@@ -84,6 +84,7 @@ Standard library changes
 * `push!(c::Channel, v)` now returns channel `c`. Previously, it returned the pushed value `v` ([#34202]).
 * `RegexMatch` objects can now be probed for whether a named capture group exists within it through `haskey()` ([#36717]).
 * A new standard library `TOML` has been added for parsing and printing [TOML files](https://toml.io) ([#37034]).
+* `RegexMatch` now iterate to give their captures. This means you can write `foo(::Noting)` and `foo((cap1, cap2)::RegexMatch)` to dispatch on if `match` found anything or not ([#34355]).
```

Comment:
```suggestion
* `RegexMatch` now iterate to give their captures. This means you can write `foo(::Nothing)` and `foo((cap1, cap2)::RegexMatch)` to dispatch on if `match` found anything or not ([#34355]).
```

Reply:
This brings up the thought of having `foo((cap1, cap2)::typeof(regex))` to allow for dispatch on multiple regular expressions (this would require parameterizing `RegexMatch`)

Reply:
You mean like if we have `RegexMatch{r"This is (\d) regex"i}`?
That would be super cute, i can see that that would be handy for things like parsing a string with a list of regexs that might match.
After this is merged can open an issue about it.

Reply:
Why did you mark this as resolved without fixing it?

Reply:
> Why did you mark this as resolved without fixing it?

Not indentional.
GitHub automatically marks it as resolved when you click apply suggestion.
Top comment has a suggestion to fix a typo.